squire_lib/src/ffi_tournament.rs

```diff
@@ -86,6 +123,173 @@ impl TournamentId {
         }
     }
 
+    /// Updates the settings
+    /// Values that are not to be changed should remain the
+    /// current setting, that would be the value the user
+    /// selected in the GUI so that is fine.
+    /// All input must be non-null.
+    ///
+    /// If any errors occur then all actions are rolled back
+    /// and, false returned.
+    ///
+    /// Otherwise true is returned and the operations are all
+    /// applied to the tournament.
+    #[no_mangle]
+    pub extern "C" fn tid_update_settings(
+        self,
+        tid: TournamentId,
+        __format: *const c_char,
+        starting_table_number: u64,
+        use_table_number: bool,
+        game_size: u8,
+        min_deck_count: u8,
+        max_deck_count: u8,
+        reg_open: bool,
+        require_check_in: bool,
+        require_deck_reg: bool,
+    ) -> bool {
+        let mut tournament: &Tournament;
+        unsafe {
+            match FFI_TOURNAMENT_REGISTRY.get().unwrap().get_mut(&self) {
+                Some(mut v) => tournament = v.value(),
+                None => {
+                    println!("[FFI]: Cannot find tournament in update_settings");
+                    return false;
+                }
+            }
+        }
+
+        // Sort input strings out
+        let format =
+            String::from(unsafe { CStr::from_ptr(__format).to_str().unwrap().to_string() });
+
+        // Init list of operations to execute
+        let aid: AdminId = Uuid::default().into();
+        let mut op_vect: Vec<TournOp> = Vec::<TournOp>::new();
+        if format != tournament.format {
```

Comment:
Strictly speaking, you don't need to check for equality here. No tournament setting will return an error if the new value is equal to the old value.

Reply:
Will it add an operation that changes nothing to the log though?

Reply:
Yes, which shouldn't be a massive problem.